gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/control/HeapMemoryMonitor.java

```diff
@@ -573,37 +573,56 @@ public final void updateStateAndSendEvent() {
     updateStateAndSendEvent(getBytesUsed());
   }
 
+  private static final SystemProperties sysProps = SystemProperties
+      .getServerInstance();
+  public static boolean DELAY_MEMORY_EVENT = sysProps.getBoolean(
+      "delay_memory_event", true);
+  public static final int MAX_DELAY_COUNT = sysProps.getInteger(
+      "delay_memory_event", 5);
+
+  private long prevBytesUsed = 0;
+  private int countSinceUpsurge = 0;
+  private long upsurgeStartTime = 0;
```

Comment:
Can these have sync issues, or we know that all methods here are invoked in the same single thread?